File: src/api/providers.ts
import { invoke } from '@tauri-apps/api/core'

export interface ProviderInfo {
  name: string
  configured: boolean
  testOk: boolean
}

export async function listProviders(): Promise<ProviderInfo[]> {
  return invoke<ProviderInfo[]>('cmd_providers_list')
}

export async function testProvider(provider: string): Promise<boolean> {
  return invoke<boolean>('cmd_providers_test', { provider })
}

File: src-tauri/src/ipc/providers.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProviderInfo {
    pub name: String,
    pub configured: bool,
    pub test_ok: bool,
}

#[tauri::command]
pub async fn cmd_providers_list() -> Result<Vec<ProviderInfo>, String> {
    // List all ASR/LLM providers with their status
    Ok(vec![])
}

#[tauri::command]
pub async fn cmd_providers_test(provider: String) -> Result<bool, String> {
    // Test a specific provider connection
    let _ = provider;
    Ok(true)
}
